cedar-lean/Cedar/Thm/Slicing.lean

```diff
@@ -39,18 +40,6 @@ namespace Cedar.Thm
 
 open Cedar.Spec
 
-/--
```

Comment:
Code style:  whenever possible, we prefer to have all the definitions relevant to a main theorem (like slicing here) inlined into the same module.  This makes things easier to read.  Can we put this definition back into the `Slicing` module and get rid of `SlicingDefs`?

Another reason is that Lean code style prefers short module names that don't require camel case.  In general, Lean projects put related files into folders (like how `Thm/Validation/Typechecker` is organized) rather than group them by name.  Usually, the file with definitions that are common to multiple modules will be called `Basic.lean`.  But in this case, the simplest solution is to just have everything in the single `Slicing` module.

Reply:
The reason I had to make a separate file is because putting this definition in `Slicing` makes an import cycle: `Thm/Authorization/Authorizer.lean` needs this definition, but `Slicing` needs `Thm/Authorization/Authorizer.lean`.  I could rename `SlicingDefs` to `Basic` if that would be better?

Reply:
Or I suppose we could put the definition in `Thm/Authorization/Authorizer.lean`?

Reply:
Yeah, this is a tough one ... we have an unsat set of constraints more or less :)

For the structure to look most Lean-like, we'd have everything in the same file / module, or we'd have a folder called `Slicing` that includes `Slicing/Basic.lean` with the common defs.  Everything in this folder would be imported by `Thm/Slicing.lean`.  This is what we do for `Validation`.  But it seemed like an overkill for `Slicing`.  Also, we ideally wouldn't have any cross references between `Authorization` and `Slicing` folders (i.e., we aim to have a tree).

We can't satisfy all of these constraints.

Maybe one thing to try for now is to stick `Basic.lean` in the `Authorizer` folder and put the slicing definition in there?  Maybe some other useful definition can be consolidated in that file too.  It's not ideal because I agree that "slicing" is the odd thing out here, but it may be the closest we can get to satisfying the constraints.

Reply:
I moved a bunch of the other `def`s from Slicing as well.  Let me know what you think

Reply:
Ah, looks like they're all slicing-related.  Is that right?  If so, maybe we just call it `Slicing.lean` instead of `Basic.lean` to make that clear?  I think it would be ok in terms of keeping names idiomatic.

Reply:
I also like the name `Slicing.lean`

Reply:
Renamed